climada/util/yearsets.py

```diff
@@ -0,0 +1,266 @@
+            (length = nr_resampled_years), can be reused similar to the sampling_vect
+            (to do: combine these two in one variable?)
+    OUTPUTS:
+      ais: the year impact set (ais), a struct with same fields as eis (such
+          as Value, ed, ...). All fields same content as in eis, except:
+          date(i): the year i
+          impact(i): the sum of impact for year(i).
+          frequency(i): the annual frequency, =1
+      sampling_vect: the sampling vector, technical, see code (can be used to
+                     re-create the exact same yearset)
+      """
+
+    # OLD CODE - STILL NEEDED?
+    # if years are known:
+    # events = list(EIS.calc_impact_year_set().keys())
+
+    # event_date = u_dt.date_to_str(eis.date)
+    # event_year = [i.split('-', 1)[0] for i in event_date]
+
+    #n_years_his = np.ceil(1/min(eis.frequency)).astype('int')
+    # n_events = len(eis.event_id)
+
+
+    if not year_list:
+        year_list = [str(date) + '-01-01' for date in np.arange(1, n_resampled_years+1
+                                                                ).tolist()]
+
+
+    if not np.all(eis.frequency == eis.frequency[0]):
+        LOGGER.warning("The frequencies of the single events differ among each other."
+                       "Please beware that this might influence the results.")
+
+
+    # INITIALISATION
+    # to do: add a flag to distinguish ais from other impact class objects
+    # artificially generate a generic  annual impact set (by sampling the event impact set)
+    ais = copy.deepcopy(eis)
+    ais.event_id = [] # to indicate not by event any more
+    ais.event_id = np.arange(1, n_resampled_years+1)
+    ais.frequency = [] # init, see below
+    ais.at_event = np.zeros([1, n_resampled_years+1])
+    ais.date = []
+
+
+    # sample from the given event impact set
+    if multiple_events: #multiple events per year possible
+        [impact_per_year, nr_events,
+         sampling_vect] = resample_multiple_annual_events(eis, n_resampled_years,
+                                                          nr_events_per_year, sampling_vect=None)
+    else: #for hazards such as RC where there is exactly one event every year
+        [impact_per_year, nr_events,
+         sampling_vect] = resample_single_annual_event(eis, n_resampled_years,
+                                                       sampling_vect)
+
+
+    #adjust for sampling error
+    correction_factor = calculate_correction_fac(impact_per_year, n_resampled_years, eis)
+    # if correction_factor > 0.1:
+    #     tex = raw_input("Do you want to exclude small events?")
+
+
+    ais.date = u_dt.str_to_date(year_list)
+    ais.at_event = impact_per_year / correction_factor
+    ais.frequency = np.ones(n_resampled_years)*sum(nr_events)/n_resampled_years
+
+
+    return ais, sampling_vect, nr_events
+
+def resample_single_annual_event(eis, nr_resampled_years, sampling_vect=None):
```

Comment:
Why use the words `resample` if the method is only `sampling` ? idem for `resample_multiple_annual_events`.